### ISSUE TITLE
New quest: What types of vehicles can be charged here?

### ISSUE THREAD

**Issue by SLMapper** [streetcomplete/StreetComplete#4003]
### General
Affected tag to be modified: [charging_station](https://wiki.openstreetmap.org/wiki/Key:charging_station)
Question asked: **What types of vehicles can be charged here?**

### Checklist
Checklist for quest suggestions (see [guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)):
- [x] 🚧 To be added tag is established and has a useful purpose
- [x] 🤔 Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)
- [x] 🐿️ Easily answerable by everyone from the outside but a survey is necessary
- [x] 💤 Not an overwhelming percentage of quests have the same answer (No spam)
- [x] 🕓 Applies to a reasonable number of map data (Worth the effort)

#### Reasonability
_around germany [overpass] (world wide [taginfo])_:
- All charging stations: 32.673 ([67.843](https://taginfo.openstreetmap.org/tags/amenity=charging_station#overview))
- Charging stations for cars: [10.730](http://overpass-turbo.eu/?Q=node%5Bamenity%3D%22charging_station%22%5D%5Bmotorcar%3Dyes%5D(%7B%7Bbbox%7D%7D)%3B%0Aout%3B&C=50.8961;9.45923;6), ([18.849](https://taginfo.openstreetmap.org/tags/amenity=charging_station#combinations))
- Charging stations for bikes: [3.622](http://overpass-turbo.eu/?Q=node%5Bamenity%3D%22charging_station%22%5D%5Bbicycle%3Dyes%5D(%7B%7Bbbox%7D%7D)%3B%0Aout%3B&C=50.8961;9.45923;6), ([4.031](https://taginfo.openstreetmap.org/tags/amenity=charging_station#combinations))
- Charging stations without vehicle information: [18.449](http://overpass-turbo.eu/?Q=node%5Bamenity%3D%22charging_station%22%5D%5B!motorcar%5D%5B!bicycle%5D%5B!scooter%5D%5B!motorcycle%5D%5B!truck%5D(%7B%7Bbbox%7D%7D)%3B%0Aout%3B&C=50.8961;9.45923;6), (>40.770)

see examples for bicycle charging stations on the [German wiki page](https://wiki.openstreetmap.org/wiki/DE:Tag:amenity%3Dcharging_station)

### Ideas for implementation

**Element selection:** 
```
    override val elementFilter = """
        nodes, ways with
          amenity = charging_station
          and !bicycle and !scooter and !motorcar and !motorcycle and !truck
    """
```

**Proposed UI:**
- Image selection with icons for: <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c1/Sinnbild_Radfahrer%2C_StVO_1992.svg/300px-Sinnbild_Radfahrer%2C_StVO_1992.svg.png" width="40" alt="bicycle"/> bicycle, <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/Sinnbild_PKW.svg/300px-Sinnbild_PKW.svg.png" width="60" alt="car"/> motorcar, <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/7/7e/Sinnbild_Kraftrad.svg/300px-Sinnbild_Kraftrad.svg.png" width="40" alt="motorcycle"/> motorcycle, <img src="https://www.shareicon.net/data/2015/12/28/694680_transport_512x512.png" width="30" alt="scooter"/> scooter, <img src="https://upload.wikimedia.org/wikipedia/commons/archive/8/84/20220311112855%21Sinnbild_LKW.svg" width="60" alt="truck"/> truck
(multi select possible)
- Cannot be answered (hide / leave note)

**Related issues and people potentially interested:**:
- https://github.com/streetcomplete/StreetComplete/issues/899 (imho outdated argument, see numbers above)
- https://github.com/streetcomplete/StreetComplete/issues/117#issuecomment-656587285
- @RubenKelevra @tiptoptom @matkoniecz @stefan005 @westnordost

**Comment by RubenKelevra** [comment]
I feel like we don't need a distinction between E-Scooters and bicycles, as I don't think there are any in the real world.

We just want to make sure to clearly distinct between these two types of "e-scooters". One is rated as small motorcycle and one is rated more like a bike. So if the charging station is on a cycle way you cannot charge a motorcycle on it:

![51KMIvfWgEL _AC_SY580_](https://user-images.githubusercontent.com/614929/166234526-d67f6248-f419-436c-8a81-ce3a8e2f8e27.jpg)
![e-roller-niu-m1pro-45-km-h-weiss-einsitzer-neu](https://user-images.githubusercontent.com/614929/166234528-c32c3e81-68b4-4fef-b8af-db56e9e584b7.jpg)

**Comment by matkoniecz** [comment]
How one may distinguish whether given charging station is for cars or for cars and trucks?

Is it about vehicles such as https://en.wikipedia.org/wiki/File:Actros182201.jpg ? Are there any such vehicles in operation which are powered by electricity?

**Comment by RubenKelevra** [comment]
@matkoniecz good question. I feel like there shouldn't be any distinguishing. 

Issue is: There's no such thing as a single tag for "general two double tracked motor vehicles which are like cars" as [double_tracked_motor_vehicle](https://wiki.openstreetmap.org/wiki/Key:double_tracked_motor_vehicle)=* is not widely used or got any vote-based approval.

[motorcar](https://wiki.openstreetmap.org/wiki/Key:motorcar)=* is used instead with a bunch of other tags, as motorcar alone doesn't even include [goods](https://wiki.openstreetmap.org/wiki/Key:goods)=*

The only option to avoid this would be to set [motor_vehicle](https://wiki.openstreetmap.org/wiki/Key:motor_vehicle) to `no` and explicitly allow [motorcycle](https://wiki.openstreetmap.org/wiki/Key:motorcycle), [moped](https://wiki.openstreetmap.org/wiki/Key:moped), [speed_pedelec](https://wiki.openstreetmap.org/wiki/Key:speed_pedelec), [mofa](https://wiki.openstreetmap.org/wiki/Key:mofa), [electric_bicycle](https://wiki.openstreetmap.org/wiki/Key:electric_bicycle), and [small_electric_vehicle](https://wiki.openstreetmap.org/wiki/Key:small_electric_vehicle).

So I guess it would better to use the documented but not-that-often used double_tracked_motor_vehicle-tag to allow or deny those vehicles.

---

So I've seen three types of charging stations:

- General purpose (with Schuko/Type2/CCS/Tesla)
- Dedicated to `electric_bicycle` which includes `electric_scooter` (since it's on a cycle track).
- Dedicated to `motorcycle` and (since cars just cannot reach them).

The questions are:
- How often is an access restriction (thus a permission) directly observable on the charging station itself?
- How regularly is the access restrictions just implied by the path leading to it – is this still an access restriction on the charging station itself?
- Would it make more sense to tag the amps + plugs and rely on the access restrictions on the path to the charging station for routers – or is it important to tag this directly for easier consumption as a POI?

**Comment by NotSoImportant** [comment]
Hi, unfortutately I see there is some uncertainity with the current tag usage already. 
I just found that even German and English wiki page lists different vehilce tags :-(
![IMG_20220503_124635](https://user-images.githubusercontent.com/35758921/166441744-ac98fd62-e9eb-447c-80e5-2c1667b878eb.jpg)
![IMG_20220503_124611](https://user-images.githubusercontent.com/35758921/166441753-bf3fb9db-6f37-4067-a620-ec6a5d23e413.jpg)

> So I've seen three types of charging stations

Yes, I more or less agree to the 3, but at least there is an additional "Dedicated to car/truck/bus" (the most common actually I guess). And there are even combinations (e.g. "2 bikes, 1 car"). 

Imho the vehicle tags should not be understood here as access restriction, but what is offered. The English wiki text about "access=no" for bus contradicts this (but in the sam way also contradicts any existing &lt;vehicle&gt;=no) 

Personal facit: as it is so unclear for some of the values, the **minimum useful implementation** I see right now is offering only the 2 by far most common (and unambigous) answers: bike ¦ motorcar. Other answers can be given by leaving a note. 
Downside: once answered the quest would not show up if in future we would add e.g. scooter. But anyhow this would already now be the case for the inconsistently mapped entries. 

Everything else will probably better be discussed in an Feature Proposal on the wiki.

**Comment by matkoniecz** [comment]
I opened https://wiki.openstreetmap.org/wiki/Talk:Tag:amenity%3Dcharging_station#trucks as it seems that it should be clarified

**Comment by SLMapper** [comment]
Thank you for your inputs.
I would go for a distinction not by acces, but by available sockets.
When looking at the above answers I only see 3 categories (+combinations of them):
- bicycle/scooter (shuko or with adapters) - _scooter not common**_
- car (any car sockets)
- truck/bus (any car sockets + enough/dedicated space - i.e. more than a regular parking slot) - _not common**_

Motorcycles or motorcycle like scooters seem [not to have different charging sockets](https://en.wikipedia.org/wiki/Electric_motorcycles_and_scooters#Charging) and therefore not require separate answer - _not common**_

** = not found in [major combinations list on taginfo](https://taginfo.openstreetmap.org/tags/amenity=charging_station#combinations), so we might omit it

### Conclusion: I see it possible to ask this quest with 2 (+1) possible answers only:
<table>
  <tr>
    <td><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/Sinnbild_PKW.svg/300px-Sinnbild_PKW.svg.png" width="60" alt="car"/><br/>car</td>
    <td><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c1/Sinnbild_Radfahrer%2C_StVO_1992.svg/300px-Sinnbild_Radfahrer%2C_StVO_1992.svg.png" width="40" alt="bicycle"/><br/>bicycle</td>
  </tr>
  <tr>
    <td colspan="2">cannot be answered</td>
  </tr>
</table>


Option 1 and 2 can either be mutli select, or another option "both" could be there.
"Cannot be answered" might e.g. be chosen because it is a potential different type.

**Comment by tiptoptom** [comment]
What would you tag if "cannot be answered" is chosen?

**Comment by SLMapper** [comment]
The same as for all other quests, that have this option already: _nothing_
Selecting this option instead offers to enter a note with two buttons: "no, hide" (this item) and "ok" -- proceeding to the "create new note" screen.

**Comment by matkoniecz** [comment]
> The same as for all other quests, that have this option already: nothing

Note that quests are expected to be able to tag all reasonable cases. Not sure how many bus/hgv/scooter/etc charging stations exist and is it OK to make impossible to tag them directly.

**Comment by SLMapper** [comment]
How is this for other quests? E.g. another type of barrier or untaggable opening hours or another material of the power pole or surface or another traffic calming or ...
For all of them the "cannot be answered" answer can be used without tagging anything.

Regarding "how many exist"?
For untagged items: who nows ;-)
For tagged items: At least as little as they do not appear in the major combinations on taglist.
To be more precise I ran **world-wide** overpass queries:
_(Side note: by far the most occurences are in Germany, Swiss, Austria, Netherlands and Belgium)_

- truck or hgv: [237](https://overpass-turbo.eu/s/1isz) &rarr; imho not relevant enough, but we could think about it
- bus: [177](https://overpass-turbo.eu/s/1isy) &rarr; imho not relevant enough
- motorcycle: [531](https://overpass-turbo.eu/s/1isx), with (!motorcar / motorcar=no): [29](https://overpass-turbo.eu/s/1isw) &rarr; not relevant
- scooter: [1173](https://overpass-turbo.eu/s/1isD) &rarr; **so probably we need to look into more details for this one**
  - with (!bicycle / bicycle=no): [625](https://overpass-turbo.eu/s/1isB)
  - with (!motorcar / motorcar=no]: [391](https://overpass-turbo.eu/s/1isC)
  - combinations of both: [134](https://overpass-turbo.eu/s/1isF)
- Undoucmented/wrong tags:
  - [1995](https://overpass-turbo.eu/s/1isE) car=yes
  - [72](https://overpass-turbo.eu/s/1isE) bike=yes

Please compare to the overall number of charging stations: [68.790](https://taginfo.openstreetmap.org/tags/amenity=charging_station#overview))

**Comment by crosspolar** [comment]
Batteryelectric Trucks/hgv will increase the next years, the new [MCS plug](https://en.wikipedia.org/wiki/Megawatt_Charging_System) has been released as standard. In Germany, the first charging stations will open next year along the A2 but other companies as Daimler and Traton proposed their own high-power charging network (again, in germany).

What I'm saying is that it might not be relevant currently but sooner or later. If you want to implement HGV-CS now or wait is up to you.

**Comment by mnalis** [comment]
(Also related: https://github.com/streetcomplete/StreetComplete/issues/903)

**Comment by georg-d** [comment]
>  truck/bus (any car sockets + enough/dedicated space - i.e. more than a regular parking slot) - _not common**_

To my understanding of above comments, a charging station suitable for truck & bus does only differ from one for motorcar by available space – but this additional space is also relevant for electric motorcars pulling a trailer, for motor homes which can be quite long despite not being a hgv, and possibly others. Hence, the amount of parties interested in truck/hgv/bus is far bigger than current or future electric truck/hgv/bus → IMHO, we shall include some answer for "long vehicles".

**Comment by NotSoImportant** [comment]
There is no such value "for long vehicles" used / documented right now.
I am wondering if we might have lost the track a bit: this should be about describing consumers of a charging station (imho a rude classification of its charging ports), not e.g. a parking lot (there is other tagging for that).

So what are the possible unique types / values based on this?

**Comment by georg-d** [comment]
> I am wondering if we might have lost the track a bit: this should be about describing consumers of a charging station (imho a rude classification of its charging ports), not e.g. a parking lot (there is other tagging for that).

Could you please give a bit more details, so other may map accordingly and create an overpass query for that? 🙂 For example, take the charging station https://www.openstreetmap.org/node/10092530762 as a typical example for the often found "station on a big parking" and how to get the information whether only a short car like smart or a long car with a trailer can be charged here. Thank you 🙂

**Comment by SLMapper** [comment]
Since we have not been able to conclude on a good StreetComplete quest definition yet, for now I have created a first [MapRoulette challenge for potential bike charging stations in Germany](https://maproulette.org/browse/challenges/29863).
This will of course only cover a small part of what was discussed here, but better than nothing.

Feel free to contribute, suggest changes or feedback ❤️

Long term I hope we will still find a good definition here and be able to add it to StreetComplete.

**Comment by matkoniecz** [comment]
https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dcharging_station#Vehicles has now claim that `hgv` tag is only about available parking space nearby

just to confirm: are electric trucks able to use all car chargers if there is sufficient space for that?

**Comment by SLMapper** [comment]
Thanks for bringing this to attention. 
Though valid, maybe its worth mentioning that it was [added by yourself @matkoniecz](https://wiki.openstreetmap.org/w/index.php?title=Special:MobileDiff/2319242&type=revision).

**Comment by matkoniecz** [comment]
Good point, I have honestly not remembered this at all (and either way it would be nice to confirm this beyond "noone removed claim from wiki page for few months")

Thanks for checking this.

**Comment by crosspolar** [comment]
@matkoniecz about electric trucks I can try help a bit: From about 2024, first heavy electric trucks will appear more on public roads while first demonstration sites (with CCS) already exist. In the beginning, trucks might be equipped with both CCS and MCS (exclusive for heavy trucks) adapters to be more flexible. With the on-going roll-out of trucks and CPs, CCS will only remain interesting for Light trucks (up to 7,5t) and MCS is likely to become standard for fast-charging of Medium and Heavy Trucks. However, overnight charging could also work off-site with smaller plugs. In the end, sites with large parking spots could end up with only one or both plugs available. There are also other ideas like on-site induction and robots, but sounds futuristic.

TLDR:
- small trucks: could use sites with CCS, but only if size of parking is sufficient
- large trucks: fast-charging with MCS, but plug-type for overnight charging not yet clear
- Technique is not fixed yet and might change in near future


Therefore, maybe the right way of asking a user is a decision tree:

```
- Plug with CCS? If yes, can hgv's be charged space-wise, especially with more than 26 tons (which would take forever to charge probably)
- Plug with MCS? --> trucks
```

or
`Is this charging site meant to be for trucks? If yes, which plugs are available?`

**Comment by westnordost** [comment]
Uff, long thread.

I really can't see a way to make this work in StreetComplete.

It is not really verifiable on the ground whether a truck could use a normal (car) charging station or not. Sure, it would probably be verifiable for those charging stations at motorway rest areas that are placed in the truck parking lot - for the simple fact that they are placed there, but not really for the rest.
Additionally, it would be spammy to ask about this for every single car charging station, it will just add a lot of errors, one way or another. (Some people would answer "yes, why not?", others would answer "no, there is no sign about it")

The only thing that remains would be to ask whether cars or whether only bicycles (or both) can be charged here. However, car chargers are in the vast majority, so vast that I (and IIRC StreetComplete) assume that if there is no explicit `car=no`, it is a car charger.
It would be spammy to ask for every single (car) charger whether also bicycles can be charged there. If there is additionally some household power plug for charging bicycles (or small scooters), I'd say, it is probably rather a brand/operator-thing, i.e. it is going to be the same for the same operator. And we have a quest to determine the charging station operator.

Bicycle-only chargers are so rare that I expect that **if** someone encounters them on a survey, he will also already add `bicycle=yes` and `car=no` or whatever the correct tag for that is. If not, still no problem because StreetComplete will ask not only about the operator but also about the capacity - in the form, a car is displayed and the question specifically asks how many cars can be charged. I.e. if it turns out that it is not a car charger at all, the user will leave a note.
